### PR TITLE
Add clear indication of non-GPU accelerated parameters in read_json docstring

### PR DESCRIPTION
**Pull request by GregoryKimball**
## Description
This PR moves the "pandas engine only" arguments to the end of the optional argument list of the docstring. 

This is the way an `admonition` will look like: 
Screen Shot 2022-10-11 at 12 06 50 PM

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by GregoryKimball** [comment]
Maybe these arguments should be marked `(pandas engine only)`?
```
typ : type of object to recover (series or frame), default 'frame'
    With cudf engine, only frame output is supported.
encoding : str, default is 'utf-8'
    The encoding to use to decode py3 bytes.
    With cudf engine, only utf-8 is supported.
```
What functionality exists here for the `cudf` engines?
``` 
compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default 'infer'
    For on-the-fly decompression of on-disk data. If 'infer', then use
    gzip, bz2, zip or xz if path_or_buf is a string ending in
    '.gz', '.bz2', '.zip', or 'xz', respectively, and no decompression
    otherwise. If using 'zip', the ZIP file must contain only one data
    file to be read in. Set to None for no decompression.
```

File: python/cudf/cudf/utils/ioutils.py [hunk 1]
```diff
@@ -473,8 +473,36 @@
 engine : {{ 'auto', 'cudf', 'cudf_experimental', 'pandas' }}, default 'auto'
     Parser engine to use. If 'auto' is passed, the engine will be
     automatically selected based on the other parameters.
-orient : string,
-    Indication of expected JSON string format (pandas engine only).
+lines : boolean, default False
+    Read the file as a json object per line.
+dtype : boolean or dict, default True
+    If True, infer dtypes, if a dict of column to dtype, then use those,
+    if False, then don't infer dtypes at all, applies only to the data.
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:480]:
```suggestion
    If True, infer dtypes for all columns; if False, don't infer dtypes at all; if a dict,
    provide a mapping from column names to their respective dtype (any missing
    columns will have their dtype inferred).
    
    
    Applies only to the data.
```

WDYT? 

I am not sure what the "applies only to the data" comment is meant to mean so I left it as a hanging sentence at the end.

File: python/cudf/cudf/utils/ioutils.py [hunk 2]
```diff
@@ -473,8 +473,36 @@
...
+byte_range : list or tuple, default None (cudf engine only)
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:481]: We could accept an iterable in the python level and tuplify it before handing off to cython, like this:
```
diff --git a/python/cudf/cudf/io/json.py b/python/cudf/cudf/io/json.py
index 2a0ae56597..5133ed1bbc 100644
--- a/python/cudf/cudf/io/json.py
+++ b/python/cudf/cudf/io/json.py
@@ -76,7 +76,7 @@ def read_json(
             dtype,
             lines,
             compression,
-            byte_range,
+            tuple(byte_range) if byte_range else None,
             engine == "cudf_experimental",
             keep_quotes,
         )
```

And then we could write:

```
byte_range : optional iterable, default None (cudf or cudf_experimental engines only)
```

File: python/cudf/cudf/utils/ioutils.py [hunk 3]
```diff
@@ -473,8 +473,36 @@
...
+    location. Reads the row that starts before or at the end of the range,
+    even if it ends after the end of the range.
+keep_quotes : bool, default False (cudf_experimental engine only)
+    If `True`, any string values are read literally (and wrapped in an
+    additional set of quotes).
+    If `False` string values are parsed into Python strings.
+typ : type of object to recover (series or frame), default 'frame'
+    With cudf engine, only frame output is supported.
+encoding : str, default is 'utf-8'
+    The encoding to use to decode py3 bytes.
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:494]: What are `py3 bytes` ?

File: python/cudf/cudf/utils/ioutils.py [hunk 4]
```diff
@@ -473,8 +473,36 @@
...
+    Read the file as a json object per line.
+dtype : boolean or dict, default True
+    If True, infer dtypes, if a dict of column to dtype, then use those,
+    if False, then don't infer dtypes at all, applies only to the data.
+byte_range : list or tuple, default None (cudf engine only)
+    Byte range within the input file to be read.
+    The first number is the offset in bytes, the second number is the range
+    size in bytes. Set the size to zero to read all data after the offset
+    location. Reads the row that starts before or at the end of the range,
+    even if it ends after the end of the range.
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:486]:
```
Pair of `(offset, length)` specifying a subrange of the file to be read, in bytes.
To read from `offset` to the end of the file, set `length=0`. Reads the starting
before or at the end of the range even if it ends past the end of the range.
```

What does "at the end of the range" mean? I guess the byte range specifies a semi-open interval `[offset, offset+length)` does that mean if a row starts at `offset + length - 1` then we read the entire row?

Aside: this is not a very ergonomic way of specifying "read from this offset to the end of the file". Could we accept either an `offset` int or an `(offset, length)` pair?

File: python/cudf/cudf/utils/ioutils.py [hunk 5]
```diff
@@ -473,8 +473,36 @@
...
+    If `False` string values are parsed into Python strings.
+typ : type of object to recover (series or frame), default 'frame'
+    With cudf engine, only frame output is supported.
+encoding : str, default is 'utf-8'
+    The encoding to use to decode py3 bytes.
+    With cudf engine, only utf-8 is supported.
+compression : {'infer', 'gzip', 'bz2', 'zip', 'xz', None}, default 'infer'
+    For on-the-fly decompression of on-disk data. If 'infer', then use
+    gzip, bz2, zip or xz if path_or_buf is a string ending in
+    '.gz', '.bz2', '.zip', or 'xz', respectively, and no decompression
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:499]: Could we not use https://pypi.org/project/python-magic/ and just detect the appropriate decompression scheme?

File: python/cudf/cudf/utils/ioutils.py [hunk 6]
```diff
@@ -473,8 +473,36 @@
...
+    otherwise. If using 'zip', the ZIP file must contain only one data
+    file to be read in. Set to None for no decompression.
+
+
```
Comment (wence-) [inline python/cudf/cudf/utils/ioutils.py:503]: Maybe put a header in here:

```
Pandas engine-specific options
-------------------------------
```
Or similar?

**Review by wence- — CHANGES_REQUESTED** [review]
Some suggestions for clarification, no need to take as is though.

File: python/cudf/cudf/utils/ioutils.py (hunk 6 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py:472]: Won't be necessary now

File: python/cudf/cudf/utils/ioutils.py (hunk 3 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py]: It a pandas.read_json only param, hence we just copied it here.

**Review by wence- — APPROVED** [review]

File: python/cudf/cudf/utils/ioutils.py (hunk 5 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py]: Do you mean we dynamically populate?

File: python/cudf/cudf/utils/ioutils.py (hunk 4 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py]: This too needs a bit of re-work like you said. I'll try to address this in https://github.com/rapidsai/cudf/issues/11780

File: python/cudf/cudf/utils/ioutils.py [hunk 7]
```diff
@@ -514,27 +533,63 @@
...
     With cudf engine, only utf-8 is supported.
 lines : boolean, default False
     Read the file as a json object per line.
 chunksize : integer, default None
-    Return JsonReader object for iteration (pandas engine only).
+
+    .. admonition:: Not GPU-accelerated
+
+       This parameter is only supported with pandas engine.
+       (i.e., ``engine='pandas'``)
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/ioutils.py:590]:
```suggestion
       This parameter is only supported with ``engine='pandas'``.
```

File: python/cudf/cudf/utils/ioutils.py [hunk 8]
```diff
@@ -547,12 +602,24 @@
     otherwise. If using 'zip', the ZIP file must contain only one data
     file to be read in. Set to None for no decompression.
 byte_range : list or tuple, default None
-    Byte range within the input file to be read (cudf engine only).
+
+    .. admonition:: GPU-accelerated
+
+       This parameter is only supported with cudf engine.
+       (i.e., ``engine='cudf'``)
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/ioutils.py:609]:
```suggestion
       This parameter is only supported with ``engine='cudf'``.
```

File: python/cudf/cudf/utils/ioutils.py [hunk 9]
```diff
@@ -547,12 +602,24 @@
...
     The first number is the offset in bytes, the second number is the range
     size in bytes. Set the size to zero to read all data after the offset
     location. Reads the row that starts before or at the end of the range,
     even if it ends after the end of the range.
 keep_quotes : bool, default False
+
+    .. admonition:: GPU-accelerated experimental feature
+
+       This parameter is only supported with cudf experimental engine.
+       (i.e., ``engine='cudf_experimental'``)
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/ioutils.py:621]:
```suggestion
       This parameter is only supported with ``engine='cudf_experimental'``.
```

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

File: python/cudf/cudf/utils/ioutils.py (hunk 5 above)
Comment (wence-) [reply python/cudf/cudf/utils/ioutils.py]: I meant that if `infer` is provided one could detect the actual file type not from the extension (brittle) but using magic (reasonably robust).

Comment (vyasr) [reply python/cudf/cudf/utils/ioutils.py]: I just read the email with this GH notification without the context of the thread. It sounded super viable.

**Pull request merged**